kiwi/xml_description.py

```diff
@@ -52,9 +54,9 @@ class XMLDescription(object):
         path to base XML description file
 
     """
-    def __init__(self, description, derived_from=None):
+    def __init__(self, description=None, derived_from=None, content=None):
         self.description_xslt_processed = NamedTemporaryFile()
-        self.description = description
+        self.description = content and BytesIO(bytes(content)) or description
```

Comment:
@aplanas tests are passing on both sides. In Salt we use `ext` from the beginning. Yet we have way less noise in the comments. :wink: